### PR TITLE
builder: use clr-installer

### PR DESCRIPTION
**Pull request by mdhorn**
Switch to using clr-installer instead of ister to generate image files.

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -480,12 +480,14 @@ func (b *Builder) copyImageTemplate(configFile, templateFile string) error {
 // updateImageBundles will update the image bundle list based on the mix bundles.
 func updateImageBundles(configValues *[]byte, bundles []string) error {
 	var data map[string]interface{}
-	err := json.Unmarshal(*configValues, &data)
+
```
Comment (reaganlo) [inline builder/builder.go]: If the new line is intentional, its fine. Else we could remove it.

File: builder/builder.go [hunk 2]
```diff
@@ -480,12 +480,14 @@ func (b *Builder) copyImageTemplate(configFile, templateFile string) error {
...
+	err := yaml.Unmarshal(*configValues, &data)
 	if err != nil {
 		return err
 	}
-	data["Bundles"] = bundles
-	*configValues, err = json.MarshalIndent(data, "", " ")
+
```
Comment (reaganlo) [inline builder/builder.go]: If the new line is intentional, its fine. Else we could remove it.

File: builder/builder.go [hunk 3]
```diff
@@ -465,10 +461,14 @@ func (b *Builder) copyImageTemplate(configFile, templateFile string) error {
 	if err == nil && len(mixBundles) > 0 {        // check if set is not empty
 		var bundles []string
 		for _, bundle := range mixBundles {
+			// Ignore additional kernel bundles
+			if strings.HasPrefix(bundle.Name, "kernel-") {
```
Comment (reaganlo) [inline builder/builder.go]: So the functional difference w.r.t to how ister worked is that,
If the user had built a mix with a different kernel like "kernel-lts", and tried to build an image with default "release-image-config.yaml", the image that gets created will have the default "kernel-native" and the "kernel-lts" would be ignored.

**Comment by rchiossi** [comment]
@mdhorn do you use that yaml library in installer? Just checking if we will need approval to use that.

**Comment by mdhorn** [comment]
> @mdhorn do you use that yaml library in installer? Just checking if we will need approval to use that.

Yes, this is the same version from the clr-installer project.

**Comment by mdhorn** [comment]
Based on discussion this afternoon I'm going to refactor this code after a new releaser of clr-installer to decouple some of the dependencies.

**Pull request closed without merging**